tensorflow/core/grappler/optimizers/meta_optimizer.cc

```diff
@@ -374,8 +374,14 @@ Status MetaOptimizer::InitializeOptimizers(
   if (AutoMixedPrecisionEnabled(cfg_.auto_mixed_precision()) &&
       AutoMixedPrecisionEnabled(
           plugin_configs.toggle_config["auto_mixed_precision"])) {
-    optimizers->push_back(
-        std::make_unique<AutoMixedPrecision>(AutoMixedPrecisionMode::CUDA));
+    if (device_types.size() == 1 &&
+        device_types.find("CPU") != device_types.end()) {
+      optimizers->push_back(
+          std::make_unique<AutoMixedPrecision>(AutoMixedPrecisionMode::FP16_CPU));
```

Comment:
I don't think we should silently run on the CPU if there is no GPU present. Probably best to create a new toggle, `auto_mixed_precision_onednn_float16`

Reply:
Addressed.